src/xref2/tools.ml

```diff
@@ -398,7 +454,7 @@ and process_module_path env ~add_canonical m p =
 and handle_module_lookup env ~add_canonical id parent sg sub =
   match Find.careful_module_in_sig sg id with
   | Some (`FModule (name, m)) ->
-      let p' = `Module (parent, name) in
+      let p' = simplify_module env (`Module (parent, name)) in
```

Comment:
I think this should be applied to the whole output of every `handle_` function so that adding a new case into the simplify function will immediately work and we won't miss any path.